### ISSUE TITLE
[Code] Recommend useful extensions (e.g. for Scala)

### ISSUE THREAD

**Issue by jankeromnes** [gitpod-io/gitpod#4155]
Gitpod currently recommends the following extensions, based on the following file extensions:

https://github.com/gitpod-io/vscode/blob/ce28a84464d2a0b45b3e7f5a1126a4276114a22d/product.json#L40-L183

However, it doesn't seem to recommend extensions for Scala files, based on the following feedback:

> One piece of feedback I might offer is having a better way to add
> additional language recognition into the editor, working with scala was a
> bit tricky as I saw no easy way to have the scala
> syntax/highlighting/coloring available for use. This might have been me not
> looking in the right spot but regardless the service worked flawlessly to
> execute anything i tried.

I think we should make sure to at least add the extensions we also recommend in https://www.gitpod.io/docs/languages-and-frameworks/

**Comment by jankeromnes** [comment]
> I think we should make sure to at least add the extensions we also recommend in https://www.gitpod.io/docs/languages-and-frameworks/

I've extracted our current language-specific extension recommendations:

- JavaScript:
  - [ ] `dbaeumer.vscode-eslint`
- Svelte:
  - [ ] `svelte.svelte-vscode`
- Python:
  - [ ] `almenon.arepl`
  - [ ] `littlefoxteam.vscode-python-test-adapter`
- Java:
  - [ ] `redhat.java`
  - [ ] `vscjava.vscode-java-debug`
  - [ ] `vscjava.vscode-java-dependency`
- C++:
  - [ ] `webfreak.debug`
- Go:
  - [ ] `ethan-reesor.vscode-go-test-adapter`
- Bash:
  - [ ] `timonwong.shellcheck`
  - [ ] `mads-hartmann.bash-ide-vscode`
- Ruby:
  - [ ] `connorshea.vscode-ruby-test-adapter`
- PHP:
  - [ ] `felixfbecker.php-debug`
- Vue:
  - [ ] `octref.vetur`
- Scala:
  - [ ] `scala-lang.scala`
  - [ ] `scalameta.metals`
- Rust:
  - [ ] `Swellaby.vscode-rust-test-adapter`
  - [ ] `serayuzgur.crates`
  - [ ] `belfz.search-crates-io`
  - [ ] `bungcip.better-toml`
- C#:
  - [ ] `muhammad-sammy.csharp`
- F#:
  - [ ] `Ionide.Ionide-fsharp`
- Dart:
  - [ ] `Dart-Code.dart-code`
- Julia:
  - [ ] `julialang.language-julia`
- LaTeX:
  - [ ] `efoerster.texlab`
  - [ ] `James-Yu.latex-workshop`
  - [ ] `JeffersonQin.latex-snippets-jeff`
- R:
  - [ ] `Ikuyadeu.r`
- Kotlin:
  - [ ] `mathiasfrohlich.Kotlin`
  - [ ] `formulahendry.code-runner`
- Deno:
  - [ ] `denoland.vscode-deno`

We may want to re-implement these as IDE recommendations (with/without notification), or to improve our recommendations.

**Comment by jankeromnes** [comment]
Also:

- Docker:
  - [ ] `ms-azuretools.vscode-docker`
  - [ ] `exiasr.hadolint`

**Comment by jankeromnes** [comment]
💡 **Idea**: Maybe aligning with (or contributing to) VS Code's own open-source recommendations would be best.

For example, they have this: https://github.com/microsoft/vscode/blob/c15cb13a383dc9ff2dc0828152e374a6b9ecc2b3/src/vs/workbench/contrib/welcome/page/browser/welcomePage.ts#L285-L298

**Comment by JanKoehnlein** [comment]
AFAIK, these are defined in the `product.json` in our [VS Code fork](https://github.com/gitpod-io/vscode/blob/c4646a76fb5abdd6e0b5a96efb1f74f5b3ed209d/product.json#L40) which we sync these recommendations with VS Code's default everytime we rebase on the latest version. To keep this step simple, I'd rather not change the VS Code defaults here.

**Comment by jankeromnes** [comment]
> AFAIK, these are defined in the `product.json` in our [VS Code fork](https://github.com/gitpod-io/vscode/blob/c4646a76fb5abdd6e0b5a96efb1f74f5b3ed209d/product.json#L40)

Indeed, as pointed out in the [first comment](https://github.com/gitpod-io/gitpod/issues/4155#issue-876694057):

> Gitpod currently recommends the following extensions, based on the following file extensions: https://github.com/gitpod-io/vscode/blob/ce28a84464d2a0b45b3e7f5a1126a4276114a22d/product.json#L40-L183

---

> which we sync these recommendations with VS Code's default everytime we rebase on the latest version. To keep this step simple, I'd rather not change the VS Code defaults here.

I agree that syncing these recommendations with VS Code is a must.

However, I don't think VS Code's own recommendations are actually open-source. For example, try searching for "extensionImportantTips" in the VS Code repository (no recommenations or `product.json` file found):

https://github.com/microsoft/vscode/search?q=extensionImportantTips

So, in order to sync with VS Code, we may have to manually try every language, and see what recommendations VS Code suggests, in order to add these to our own `product.json`.

One example: We don't currently recommend any extension for Scala, so Scala developers using Gitpod get, by default, a black-and-white code editor, with no indication how to improve this.

**Comment by akosyakov** [comment]
> So, in order to sync with VS Code, we may have to manually try every language, and see what recommendations VS Code suggests, in order to add these to our own product.json.

We take product.json from local VS Code and remove all proprietary bits.
